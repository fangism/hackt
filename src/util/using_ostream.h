/**
	\file "using_ostream.h"
	Convenient using directives for ostream, cout, cerr, endl...
	Note: this is not wrapped by include-once protection, because it may
	be used within other namespaces.  
	Also assumes that <iosfwd> was included beforehand.  
	$Id: using_ostream.h,v 1.1 2004/12/05 05:07:26 fang Exp $
 */

#ifndef	__USING_OSTREAM_H__
#define	__USING_OSTREAM_H__

using std::ostream;
using std::cout;
using std::cerr;
using std::endl;
// any others?

#endif	// __USING_OSTREAM_H__
#undef	__USING_OSTREAM_H__


