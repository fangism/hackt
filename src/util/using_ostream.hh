/**
	\file "util/using_ostream.hh"
	Convenient using directives for ostream, cout, cerr, endl...
	Note: this is not wrapped by include-once protection, because it may
	be used within other namespaces.  
	Also assumes that <iosfwd> was included beforehand.  
	$Id: using_ostream.hh,v 1.2 2005/05/10 04:51:31 fang Exp $
 */

using std::ostream;
using std::cout;
using std::cerr;
using std::endl;
// any others?

