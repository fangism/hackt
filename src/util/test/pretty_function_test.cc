/**
 */

#include <iostream>

#include "using_ostream.h"

class bar {
public:
	bar() { cerr << __PRETTY_FUNCTION__ << endl; }

	~bar() { cerr << __PRETTY_FUNCTION__ << endl; }

	void
	hello(void) const { cerr << __PRETTY_FUNCTION__ << endl; }

	static
	void
	greet(void) { cerr << __PRETTY_FUNCTION__ << endl; }

};

int
main(int argc, char* argv[]) {
	cerr << __PRETTY_FUNCTION__ << endl;
	bar bee;
	bar::greet();
	bee.hello();
	return 0;
}

