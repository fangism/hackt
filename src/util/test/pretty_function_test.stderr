int main(int, char**)
bar::bar()
static void bar::greet()
void bar::hello() const
bar::~bar()
